app/UI/message-list.tsx

```diff
@@ -405,6 +424,24 @@ function MessageActions(
                 <ReplyIcon class={`w-5 h-5 text-[#B6BAC0] hover:text-[#D9DBDE]`} />
             </button>
 
+            {(myPublicKey.hex === message.author.hex || myPublicKey.hex === relayInformation?.pubkey) &&
+                (message.event.kind === NostrKind.TEXT_NOTE) &&
+                (
+                    <button
+                        class={`flex items-center justify-center
+                p-1
+                bg-[#313338] hover:bg-[#3A3C41]`}
+                        onClick={() => {
+                            emit({
+                                type: "DeleteEvent",
+                                event: message.event,
+                            });
+                        }}
+                    >
+                        <DeleteIcon class={`w-5 h-5 text-[#B6BAC0] hover:text-[#D9DBDE]`} />
+                    </button>
+                )}
```

Comment:
Ok, now I understand why you changed the public model and pass `RelayInformation` all the way down to this layer. The goal is to know if the current user is the admin of the current relay so that deciding if `deletion button` can be shown.

For this purpose, it's better to have a single function
```
type isAdmin = (pubkey: string | PublicKey) => boolean
```
or
```
type isAdmin = (pubkey: string | PublicKey, relayURL: string | URL) => boolean
```

so that you inject a semantically strong code instead of passing general purpose data to both
1. make the code more readable
2. make the code more adaptable to changes

In general, __data structures are implementation details__ and __interfaces describe intensions__.

Reply:
Pass relayInformation has been changed to isAdmin interface.